api/envoy/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto

```diff
@@ -160,6 +160,14 @@ message UdpProxyConfig {
     // while the upstream is not ready will be dropped. In case this field is set but the options
     // are not configured, the default values will be applied as described in the ``BufferOptions``.
     BufferOptions buffer_options = 9;
+
+    // Save the response headers to the downstream info filter state for consumption
+    // by the session filters. The filter state key is ``envoy.udp_proxy.propagate_response_headers``.
+    bool propagate_response_headers = 10;
```

Comment:
How is this intended to be used? Should docs be updated?

Reply:
It's intended to be used for receiving response headers/trailers from upstream, which provides additional metadata about the upstream endpoint and connection status that can be used for telemetry purposes.
It adds similar functionality to udp_tunneling as in tcp_tunneling:
- [tcp tunneling: optionally move response header to downstream info #23118
](https://github.com/envoyproxy/envoy/pull/23118)
- [TCP Tunneling: Optionally propagate response trailers to downstream info #25023
](https://github.com/envoyproxy/envoy/pull/25023)

Reply:
Makes sense, thanks.